src/runtime_src/core/common/api/xrt_elf.cpp

```diff
@@ -28,6 +29,14 @@ class elf_impl
       throw std::runtime_error(fnm + " is not found or is not a valid ELF file");
   }
 
+  elf_impl(const std::vector<char>& data)
+  {
+    std::istringstream elf_stream;
+    elf_stream.rdbuf()->pubsetbuf(const_cast<char*>(data.data()), data.size());
```

Comment:
The const_cast looks evil here. Can you add a comment why we are doing it and the fact that elf_stream is input stream (read only) so it cannot accidentally change **data**.

Reply:
since data is const type so data.data() return const char* 
https://cplusplus.com/reference/vector/vector/data/
`If the vector object is const-qualified, the function returns a pointer to const value_type. Otherwise, it returns a pointer to value_type.`
while 
`streambuf* pubsetbuf (char* s, streamsize n);`
https://cplusplus.com/reference/streambuf/streambuf/pubsetbuf/